ext/mysql2/client.c

```diff
@@ -53,8 +53,10 @@ VALUE rb_hash_dup(VALUE other) {
  * variable to use, but MYSQL_SERVER_VERSION gives the correct numbers when
  * linking against the server itself
  */
-#ifdef LIBMYSQL_VERSION
-  #define MYSQL_LINK_VERSION LIBMYSQL_VERSION
+#ifdef MARIADB_CLIENT_VERSION_STR
+  #define MYSQL_LINK_VERSION MARIADB_CLIENT_VERSION_STR
+#elif LIBMYSQL_VERSION
+  #define MYSQL_LINK_VERSION MYSQL_VERSION_ID
```

Comment:
MYSQL_VERSION_ID is numeric and will fail the major.minor dot tests later in this file.